data/lib/core/player.lua

```diff
@@ -674,3 +664,72 @@ function Player.getTotalDefense(self)
 	end
 	return total
 end
+
+function Player.isPromoted(self)
+	return not self:getVocation():getPromotion() and true or false
+end
+
+function Player.hasBlessing(self, blessingId)
+	return self:getBlessing(blessingId) > 0
+end
+
+function Player.getBlessings(self, excludeSpecial)
+	local first = BLESSING_FIRST	
+	if excludeSpecial then
+		first = BLESSING_WISDOM_OF_SOLITUDE -- skip first two blessings
+	end
+	
+	local blessings = {}
+	for blessingId = first, BLESSING_LAST, 1 do
+		if self:hasBlessing(blessingId) then
+			blessings[#blessings + 1] = blessingId
+		end
+	end
+	return blessings
+end
+
+function Player.addBlessingsHistory(self, event, type)
+	return db.query("INSERT INTO `blessings_history` (`player_id`, `type`, `event`, `created_at`) VALUES (" .. self:getGuid() .. ", " .. type .. ", " .. db.escapeString(event) .. ", " .. os.time() .. ")")
+end
+
+function Player.getBlessingsHistory(self)
+	local history = {}
+	local resultId = db.storeQuery("SELECT `type`, `event`, `created_at` FROM `blessings_history` WHERE `player_id` = " .. self:getGuid() .. " ORDER BY `created_at` DESC")
+	if resultId then
```

Comment:
It would look cleaner if you invert this condition:

```lua
local resultId = ...
if not resultId then
  return {}
end

local history = {}
repeat
  ...
until ...
return history
```

To avoid an indent level

Reply:
yep, looks way cleaner

thanks 😁